File: HtmlPrototype/game/gameplay/playerResources.js
function playerResources(){
    var $food = 0;
    var $ore = 0;
    var $tech = 0;

    this.food = function(){
        return $food;
    };
    this.ore = function(){
        return $ore;
    };
    this.tech = function(){
        return $tech;
    };

    this.addFood = function(amount){
        $food += amount;
    };
    this.addOre = function(amount){
        $ore += amount;
    };
    this.addTech = function(amount){
        $tech += amount;
    };
}

File: StarBastard/StarBastard.Core/Gameplay/playerResources.cs
namespace StarBastard.Core.Gameplay
{
    public class PlayerResources
    {
        public int Food { get; private set; }
        public int Ore { get; private set; }
        public int Tech { get; private set; }

        public void AddOre(int amount)
        {
            Food += amount;
        }

        public void AddFood(int amount)
        {
            Ore += amount;
        }

        public void AddTech(int amount)
        {
            Tech += amount;
        }
    }
}
